### PR TITLE
CATO-4135: Adding boxes to full Creditors after one year note.

### PR DESCRIPTION
**Pull request by tmikus**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC162.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  override def validate(boxRetriever: FullAccountsBoxRetriever): Set[CtValidation] = {
+    collectErrors(
+      validateMoney(value),
+      failIf(boxRetriever.ac64().hasValue)(totalEqualToCurrentAmount(boxRetriever))
+    )
+  }
+
+  def totalEqualToCurrentAmount(boxRetriever: FullAccountsBoxRetriever)() = {
+    failIf(this.orZero != boxRetriever.ac64().orZero) {
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC162.scala]: This is a bug we have throughout calcs right now, None != Some(0), so this need to be changed to reflect that

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC163.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  override def validate(boxRetriever: FullAccountsBoxRetriever): Set[CtValidation] = {
+    collectErrors(
+      validateMoney(value),
+      failIf(boxRetriever.ac65().hasValue)(totalEqualToCurrentAmount(boxRetriever))
+    )
+  }
+
+  def totalEqualToCurrentAmount(boxRetriever: FullAccountsBoxRetriever)() = {
+    failIf(this.orZero != boxRetriever.ac65().orZero) {
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC163.scala]: As above

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/calculations/BalanceSheetCreditorsCalculator.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import uk.gov.hmrc.ct.accounts.frs102.boxes._
+import uk.gov.hmrc.ct.box.CtTypeConverters
+
+trait BalanceSheetCreditorsCalculator extends CtTypeConverters {
+
+  def calculateAC162(ac156: AC156, ac158: AC158, ac160: AC160): AC162 = {
+    (ac156.value, ac158.value, ac160.value) match {
+      case (None, None, None) => AC162(None)
+      case _ => AC162(Some(ac156.orZero + ac158.orZero + ac160.orZero))
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/calculations/BalanceSheetCreditorsCalculator.scala]: The implicit convertors should do the .orZero for you

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/calculations/BalanceSheetCreditorsCalculator.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  }
+
+  def calculateAC163(ac157: AC157, ac159: AC159, ac161: AC161): AC163 = {
+    (ac157.value, ac159.value, ac161.value) match {
+      case (None, None, None) => AC163(None)
+      case _ => AC163(Some(ac157.orZero + ac159.orZero + ac161.orZero))
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/calculations/BalanceSheetCreditorsCalculator.scala]: As above

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC162.scala (hunk 1 above)
Comment (tmikus) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC162.scala]: Is there an example of that fix?

**Pull request merged**